pkg/sql/testdata/logic_test/alter_table

```diff
@@ -423,6 +423,12 @@ ALTER TABLE add_default ALTER COLUMN b DROP DEFAULT
 statement error null value in column "b" violates not-null constraint
 INSERT INTO add_default (a) VALUES (4)
 
+statement ok
+ALTER TABLE add_default ALTER COLUMN b SET DEFAULT NULL
+
+statement error null value in column "b" violates not-null constraint
```

Comment:
Because the `INSERT` doesn't specify a value for `b`, so it uses the default value which is `NULL`.